Firestore/core/src/firebase/firestore/nanopb/nanopb_util.h

```diff
@@ -88,6 +90,20 @@ inline std::vector<uint8_t> MakeVector(const ByteString& str) {
   return {str.begin(), str.end()};
 }
 
+/**
+ * Due to the nanopb implementation, nanopb_boolean could be an integer
+ * other than 0 or 1, (such as 2). This leads to undefined behaviour when
+ * it's read as a boolean. eg. on at least gcc, the value is treated as
+ * both true *and* false. So we'll instead memcpy to an integer (via
+ * absl::bit_cast) and compare with 0.
+ *
+ * Note that it is necessary to pass-by-reference here to get the original
+ * value of `nanopb_boolean`.
+ */
+inline bool ReadBoolean(const bool& nanopb_boolean) {
```

Comment:
`bool` is a primitive type so you can just pass it by value (i.e. just `bool` instead of `const bool&`).

Also consider naming this something like `SafeReadBoolean` to help indicate that this is defensive.

Reply:
Renamed.

It has to be a `const bool&` though..pass-by-value here is subject to the same issue that this might be both true and false. `TEST_F(SerializerTest, BadBoolValueInterpretedAsTrue)` fails with pass-by-value here, because in `SafeReadBoolean`, the passed in value is `false` already.

Reply:
Makes sense. Thanks for clarifying.